[AC-5484] Add home program bin

jonkiparsky opened the pull request:
Probably some improvements to make, both in the new code and in the base code. Maybe do those pairing?

jonkiparsky committed:
[AC-5484] Add home program bin

jonkiparsky committed:
[AC-5484] Typo in README

jonkiparsky committed:
[AC-5484] Add __init__.py to make python recognize the module

mc-nathan commented on allocate.py:
```diff
@@ -54,9 +56,12 @@ def work_left(bins):
                  for value in industry.values]
+program_bins = [HomeProgramBin(program=value,
+                               value=3*BIN_DEFAULT)
+                for value, _ in program.values]
```
I'd make these parallel.  I like yours slightly better so let's do that.

mc-nathan commented on classes/entity.py:
```diff
@@ -9,7 +9,7 @@
                "completed"]
 CSV_HEADER = ",".join(CSV_COLUMNS)
-CSV_FORMAT = "{%s}" % "},{".join(CSV_COLUMNS) + "}"
+CSV_FORMAT = "{%s}" % ("},{".join(CSV_COLUMNS))
```
Good catch.

mc-nathan commented on classes/home_program_bin.py:
```diff
@@ -0,0 +1,28 @@
+        return startup.properties['program'] == self.program
+
+    def value(self, judge):
+        if judge.properties['industry'] == self.program:
```
s/industry/program

jonkiparsky replied:
doh! caught in copy-paste!

mc-nathan commented on classes/home_program_bin.py:
```diff
@@ -0,0 +1,28 @@
+        if judge.properties['industry'] == self.program:
+            return super().value(judge)
+        else:
+            #None? mixed-type returns are ugly
```
Could do `return BIN_NO_VALUE` from the bin.py.

mc-nathan commented on classes/judge.py:
```diff
@@ -17,7 +17,7 @@ def __init__(self, data=None):
         for property in Property.all_properties:
             self.add_property(property, data)
-        self.remaining = int(self.properties.get("commitment", 50))
+        self.remaining = int(self.properties.get("commitment", 50) or 0) 
```
What is the `or 0` doing?

jonkiparsky replied:
I had a failure where it tried to make "" into an int. This is a quick way to hack around that.

mc-nathan replied:
Looks like the generator script is broken.  I'll get that fixed.

jonkiparsky committed:
[AC-5484] Minor changes from code review

mc-nathan committed:
[AC-5484] Switch from values to weights
- Add bin_factory function in allocate.py
- Switch default number of reads 4 so it works with example.csv

mc-nathan committed:
Fix issue with cutoff test
